leases/schema/utils.py

```diff
@@ -10,3 +17,41 @@ def parse_to_dict(object):
         return {"id": to_global_id(node_type, id), "error": error}
 
     return parse_to_dict
+
+
+def lookup_or_create_boat(info, input: dict) -> Optional[Boat]:
```

Comment:
The related case VEN-1271 (script to create boats for processed applications) describes this matching logic::

> If the customer already has a boat with the same registration number, update the fields of the existing boat. The information in the application is assumed to be more accurate.

The same logic would apply here?

Reply:
Good point, this is definitely a simplified version, but maybe it's worth investing a bit on making it a bit more robust. 👍

Reply:
Added